bin/csv_output.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+import sys
+import csv
+import json
+
+
+def generate_output_dict(input: str) -> dict:
+    """
+    creates template output_dict from input csv file and including other necessary fields
+
+    Args:
+        input (str): file path to input csv file
+
+    Returns:
+        dict: dictionary containing data from input dict and new fields needed for output csv
+    """
+    output_dict = {}
+    with open(input) as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            output_dict.update(
+                {
+                    row["synapse_id"]: {
+                        "synapse_id": row["synapse_id"],
+                        "file_type": row["file_type"],
+                        "md5_checksum": row["md5_checksum"],
+                        "version_number": "",
+                        "required_tests": [],  # placeholder for ome.tiff until we have required_tests from input
+                        "is_valid": "",
+                        "passed_tests": [],
+                        "failed_tests": [],
+                        "skipped_tests": [],  # placeholder for future version with skipping
+                    }
+                }
+            )
+    return output_dict
+
+
+def update_output_dict(json_list: list, output_dict: dict) -> dict:
+    """
+    updates output dict with validation results from all JSON files.
+    compares passed_tests to required_tests to determine if file is valid.
+
+    Args:
+        json_list(list): list of JSON validation files
+        output_dict (dict): output_dict generated to contain input.csv data and validation result data
+
+    Returns:
+        dict: updated output_dict with validation test results
+    """
+    # add validation results from JSON
+    for item in json_list:
+        # read validation data from json
+        with open(item, "r") as file:
+            data = json.load(file)
+            # save reused variables
+            syn_id = data["input"]["synapse_id"]
+            test_name = data["test"]["name"]
+            # set version number in output_dict
+            output_dict[syn_id]["version_number"] = data["input"]["version_number"]
+            # append required_tests
+            output_dict[syn_id]["required_tests"].append(test_name)
+            # add tests to passed_tests or failed_tests
+            if data["test"]["status"] == "pass":
+                output_dict[syn_id]["passed_tests"].append(test_name)
+            else:
+                output_dict[syn_id]["failed_tests"].append(test_name)
+
+    # set is_valid
+    for k in output_dict:
+        if set(output_dict[k]["passed_tests"]).issuperset(
+            set(output_dict[k]["required_tests"])
+        ):
+            is_valid = True
+        else:
+            is_valid = False
+        output_dict[k]["is_valid"] = is_valid
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** By using the `.items()` dict method, you can streamline this for-loop and if-else statement as follows:

```suggestion
    for k, v in output_dict.items():
        v["is_valid"] = set(v["passed_tests"]) >= set(v["required_tests"])
```